network/common/src/main/java/org/apache/spark/network/client/TransportClient.java

```diff
@@ -85,6 +86,24 @@ public SocketAddress getSocketAddress() {
   }
 
   /**
+   * Returns the ID used by the client to authenticate itself when authentication is enabled.
+   *
+   * @return The client ID.
+   */
+  public String getClientId() {
+    return clientId;
+  }
+
+  /**
+   * Sets the authenticated client ID. This is meant to be used by the authentication layer;
+   * trying to set a different client ID after it's been set will result in an exception.
+   */
+  public void setClientId(String id) {
+    Preconditions.checkState(clientId == null, "Client ID has already been set.");
```

Comment:
If I understand correctly, this will never get called when you aren't using authentication, right?  Maybe drop in a comment here explaining that, and on `getClientId` that it will return `null` without authentication?